ui/app/helpers/utils/conversion-util.js

```diff
@@ -60,60 +58,75 @@ const baseChange = {
-const invertConversionRate = (conversionRate) => () => new BigNumber(1.0).div(conversionRate)
-const decToBigNumberViaString = () => R.pipe(String, toBigNumber['dec'])
-
-// Predicates
-const fromAndToCurrencyPropsNotEqual = R.compose(
-  R.not,
-  R.eqBy(R.__, 'fromCurrency', 'toCurrency'),
-  R.flip(R.prop)
-)
-
-// Lens
-const valuePropertyLens = R.over(R.lensProp('value'))
-const conversionRateLens = R.over(R.lensProp('conversionRate'))
-
-// conditional conversionRate setting wrapper
-const whenPredSetCRWithPropAndSetter = (pred, prop, setter) => R.when(
-  pred,
-  R.converge(
-    conversionRateLens,
-    [R.pipe(R.prop(prop), setter), R.identity]
-  )
-)
-
-// conditional 'value' setting wrappers
-const whenPredSetWithPropAndSetter = (pred, prop, setter) => R.when(
-  pred,
-  R.converge(
-    valuePropertyLens,
-    [R.pipe(R.prop(prop), setter), R.identity]
-  )
-)
-const whenPropApplySetterMap = (prop, setterMap) => whenPredSetWithPropAndSetter(
-  R.prop(prop),
-  prop,
-  R.prop(R.__, setterMap)
-)
-
-// Conversion utility function
-const converter = R.pipe(
-  whenPredSetCRWithPropAndSetter(R.prop('conversionRate'), 'conversionRate', decToBigNumberViaString),
-  whenPredSetCRWithPropAndSetter(R.prop('invertConversionRate'), 'conversionRate', invertConversionRate),
-  whenPropApplySetterMap('fromNumericBase', toBigNumber),
-  whenPropApplySetterMap('fromDenomination', toNormalizedDenomination),
-  whenPredSetWithPropAndSetter(fromAndToCurrencyPropsNotEqual, 'conversionRate', convert),
-  whenPropApplySetterMap('toDenomination', toSpecifiedDenomination),
-  whenPredSetWithPropAndSetter(R.prop('numberOfDecimals'), 'numberOfDecimals', round),
-  whenPredSetWithPropAndSetter(R.prop('roundDown'), 'roundDown', roundDown),
-  whenPropApplySetterMap('toNumericBase', baseChange),
-  R.view(R.lensProp('value'))
-)
+/**
+ * Defines the base type of numeric value
+ * @typedef {('hex' | 'dec' | 'BN')} NumericBase
+ */
+
+/**
+ * Defines which type of denomination a value is in
+ * @typedef {('WEI' | 'GWEI' | 'ETH')} EthDenomination
+ */
+
+/**
+ * Utility method to convert a value between denominations, formats and currencies.
+ * @param {Object} input
+ * @param {string | BigNumber} input.value
+ * @param {NumericBase} input.fromNumericBase
+ * @param {EthDenomination} [input.fromDenomination]
+ * @param {string} [input.fromCurrency]
+ * @param {NumericBase} input.toNumericBase
+ * @param {EthDenomination} [input.toDenomination]
+ * @param {string} [input.toCurrency]
+ * @param {number} [input.numberOfDecimals]
+ * @param {number} [input.conversionRate]
+ * @param {boolean} [input.invertConversionRate]
+ * @param {string} [input.roundDown]
+ */
+const converter = ({
+  value,
+  fromNumericBase,
+  fromDenomination,
+  fromCurrency,
+  toNumericBase,
+  toDenomination,
+  toCurrency,
+  numberOfDecimals,
+  conversionRate,
+  invertConversionRate,
+  roundDown,
+}) => {
+  let convertedValue = toBigNumber[fromNumericBase ?? 'dec'](value)
```

Comment:
This seems different from the old implementation. The old version did something like this:

```suggestion
  let convertedValue = fromNumericBase ? toBigNumber[fromNumericBase](value) : value
```

In most cases, I think defaulting to `dec` would make sense, since most of these parameters depend upon `value` being a `BigNumber` instance and would blow up otherwise. And even if `value` is already a `BigNumber` when it's passed in, converting it again is harmless enough (`BigNumber` defaults to being stringified as base-10 if not specified, so `dec` is the appropriate default).

There is one case I've found so far where this might change the behavior though: if only `value` and `toNumericBase` are provided. In that case, this would give a different result than before. For example, the input `{ value: '16', toNumericBase: 'hex' }` would give the result `16` before, whereas now it'd give `10`.

Reply:
The old version would only convert the value to a big number if there was a provided fromNumericBase, however the way the rest of the functions were written it expected the value to already be a big number. I put the logic behind a if check `(if fromNumericBase, toBigNumber)` and ran your sample case... 

With `value: 16` the result is `10` with `value: '16'` the result is `'16'`.. Are we expecting value to always be a string?

Reply:
Good question. I have no idea. It was difficult to tell due to indirection

Reply:
if I throw an error when fromNumericBase is not supplied over 50 tests blow up. Sooooo... I'm gonna just put the conversion behind a logic check for this iteration. <_<